scripts/release-checks/dependency-ranges/build-angular-peer-deps-check.mts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import path from 'path';
+import url from 'url';
+import semver from 'semver';
+import { loadPackageJson } from './pkg-json.mjs';
+
+/** Path to the current directory. */
+const currentDir = path.dirname(url.fileURLToPath(import.meta.url));
+
+/** Path to the project directory. */
+const projectDir = path.join(currentDir, '../../../');
+
+export async function checkBuildAngularPeerDependencies(
+  newVersion: semver.SemVer,
+): Promise<string[]> {
+  const pkgJson = await loadPackageJson(
+    path.join(projectDir, 'packages/angular_devkit/build_angular/package.json'),
+  );
+  const { major, minor } = newVersion;
+  const isPrerelease = !!newVersion.prerelease[0];
+  const isMajor = minor === 0;
+  const failures: string[] = [];
+
+  let expectedFwPeerDep = `^${major}.0.0`;
+  if (isMajor && isPrerelease) {
+    expectedFwPeerDep = `^${major}.0.0-next.0`;
+  } else if (isPrerelease) {
+    expectedFwPeerDep = `^${major}.0.0 || ^${major}.${minor}.0-next.0`;
+  }
+
+  for (const [depName, range] of Object.entries(pkgJson.peerDependencies)) {
+    // Even though `ng-packagr` might not strictly follow the same release schedules
+    // like official Angular packages, we generally expect it to match. It's better
+    // flagging it than silently passing pre-checks. The caretaker can always forcibly
+    // ignore this check.
+    if (!depName.startsWith('@angular/') && depName !== 'ng-packagr') {
+      continue;
+    }
+
+    if (range !== expectedFwPeerDep) {
+      failures.push(
+        `${pkgJson.name}: Unexpected peer dependency for "${depName}". Expected: ${expectedFwPeerDep}`,
```

Comment:
Would be helpful to indicate which package.json caused the failure.

Reply:
I decided to not do this because the package name should be sufficient. Feel free to add this in the future if it ends up being confusing.